flows/index.py

```diff
@@ -526,27 +533,46 @@ def get_updated_passage_concepts(
     It is also, not possible to duplicate a Concept object in the concepts array as we
     are removing all instances where the model is the same.
     """
-    if not passage.concepts:
-        return [concept.model_dump(mode="json") for concept in concepts]
+    match change:
```

Comment:
suggestion: When we implement the work we split into two or more PR's

Reply:
Resolving as also brought up on [Slack](https://climate-policy-radar.slack.com/archives/C08F40KPLNR/p1741279239592469?thread_ts=1741193276.013729&cid=C08F40KPLNR).